app/packages/core/src/plugins/SchemaIO/components/DateTimeView.tsx

```diff
@@ -0,0 +1,71 @@
+import * as fos from "@fiftyone/state";
+import {
+  dateFromDateString,
+  dateFromDateTimeString,
+  formatDatePicker,
+  formatDateTimePicker,
+  INPUT_TYPE_DATE,
+  INPUT_TYPE_DATE_TIME,
+  styles,
+} from "@fiftyone/utilities";
+import React, { useEffect, useMemo, useState } from "react";
+import { useRecoilValue } from "recoil";
+import { useKey } from "../hooks";
+import { ViewPropsType } from "../utils/types";
+import FieldWrapper from "./FieldWrapper";
+
+export default function DateTimeView(props: ViewPropsType) {
+  const timeZone = useRecoilValue<string>(fos.timeZone);
+  const [formattedDate, setFormattedDate] = useState<string>("");
+  const { onChange, schema, path, data } = props;
+  const { compact, placeholder = "", readOnly } = schema.view;
+  const [key, setUserChanged] = useKey(path, schema, data, true);
+
+  const dateOnly = schema.view.date_only;
+  const inputType = dateOnly ? INPUT_TYPE_DATE : INPUT_TYPE_DATE_TIME;
+
+  const parseTimestamp = useMemo(() => {
+    return dateOnly
+      ? dateFromDateString
+      : (v: string) => dateFromDateTimeString(timeZone, v);
+  }, [dateOnly, timeZone]);
+  const formatDate = useMemo(() => {
+    return dateOnly
+      ? formatDatePicker
+      : (v: string) => formatDateTimePicker(timeZone, v);
+  }, [dateOnly, timeZone]);
+
+  useEffect(() => {
+    if (!data) {
+      setFormattedDate("");
+      return;
+    }
+
+    // populate the user-facing field with data from the backend
+    setFormattedDate(formatDate(data));
```

Comment:
Based on the python definition in `types.py`, `data` should be timestamp (number) not a string. `formatDatePicker` method's type definition has a small bug.